pp_ctl.c

```diff
@@ -4845,8 +4846,54 @@ PP(pp_leaveeval)
     oldsp = PL_stack_base + cx->blk_oldsp;
     gimme = cx->blk_gimme;
 
-    /* did require return a false value? */
-    failed =    CxOLD_OP_TYPE(cx) == OP_REQUIRE
+    bool is_require= CxOLD_OP_TYPE(cx) == OP_REQUIRE;
+    if (is_require) {
+        /* We are in an require. Check if use feature 'module_true' is enabled,
+         * and if so later on correct any returns from the require. */
+        COP *old_pl_curcop = PL_curcop;
+        OP *check = PL_op;
+
+        /* we might be called for an OP_LEAVEEVAL or OP_RETURN opcode
+         * and the parse tree will look different for either case.
+         * so find the right op to check later */
+        if ((check->op_flags & OPf_KIDS) && OP_TYPE_IS_OR_WAS(check, OP_RETURN)) {
+            /* it is a return statement, so we need scan upwards until we find an
+             * OP_LEAVEEVAL, consider (0, return 0) */
+            do {
+                check= op_parent(check);
+            } while (check && !OP_TYPE_IS_OR_WAS(check, OP_LEAVEEVAL));
+        }
+        if (check && (check->op_flags & OPf_KIDS) && OP_TYPE_IS_OR_WAS(check, OP_LEAVEEVAL)){
+            check = cUNOPx(check)->op_first;
+
+            /* ok, we found something to check, we need to scan through
+             * it and an OP_NEXTSTATE and then read the feature state out
+             * of it. It isn't clear if we can stop after the firt OP_NEXTSTATE
+             * or not so we scan through until we run out of things to check
+             * or find something that is true. */
+            if (check) {
+                const OP *kid = cLISTOPx(check)->op_first;
+                if (0) op_dump(kid);
```

Comment:
leftover debugging code, I assume

Reply:
Yep, will remove.

Reply:
Well, maybe not yet, its still useful for debugging. :-)